pyk/src/pyk/ktool/krun.py

```diff
@@ -129,8 +133,13 @@ def run(
             pipe_stderr=pipe_stderr,
             bug_report=bug_report,
             debugger=debugger,
+            proof_hint=proof_hint,
         )
 
+        if proof_hint:
+            # We didn't pipe the result to stdout, so it was already printed 
+            return
```

Comment:
This is a new mode of operation, please add a dedicated function, say, `run_proof_hint` for this.

Reply:
This isn't a comment about your option 3.

Reply:
What I wrote:

> I've left a few comments for the current approach (i.e. (3) from the list).

What I meant: either go with the current approach (option 3) and fix the following, or do option 1.